server/server.go

```diff
@@ -53,6 +54,8 @@ type Server struct {
 	routinesPool                  *safe.Pool
 	leadership                    *cluster.Leadership
 	defaultForwardingRoundTripper http.RoundTripper
+	metricsRegistry               metrics.Registry
+	metricsEnabled                bool
```

Comment:
I double checked the implementation and my described approach above won't work. All different registries that we create through e.g. `NewVoidRegistry`, `RegisterStatsd`, `RegisterPrometheus`.. will have the internal type `standardRegistry` and therefore I can't use a type assertion.

This means a `metricsEnabled()` utility would have to check on the global configuration once more. E.g. `if globalConfiguration.Web.Metrics.Prometheus != nil || globalConfiguration.Web.Metrics.Statsd != nil ...`. Personally I think that having this conditions only in `registerMetricClients` is the more clean approach and tend to leave it like it is. WDYT?